torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -1454,6 +1454,11 @@ class ShapePropagator {
         node->output()->setType(weight_type->withDim(indices_type->dim() + 1));
         return true;
       }
+    } else if (node->matches("aten::linear(Tensor input, Tensor weight, Tensor? bias) -> Tensor")) {
+      if (auto type = input_type(0)) {
+        node->output()->setType(type);
```

Comment:
is `type` here a complete tensor or just a dimensioned tensor?

the input shape is not the same as the output shape right?

Reply:
hmm you are right, actually it should be a complete tensor and the output shape should be different. seems like bilinear below is also wrong, will fix it.